### PR TITLE
Wip fix key perms

### PR DESCRIPTION
**Pull request by oms4suse**
Fix keyring permissions and use the rados_client inherited object more.

Signed-off-by: Owen Synge osynge@suse.com

### CODE REVIEW

File: ceph_cfg/rados_client.py [hunk 1]
```diff
@@ -100,3 +113,86 @@ def create(self):
...
+        keyringobj.key_type = self.bootstrap_keyring_type
+        oldmask = os.umask(077)
+        try:
+            try:
+                tmpd = tempfile.mkdtemp()
+                key_path = os.path.join(tmpd, "keyring")
+                arguments_prefix = [
+                    util_which.which_ceph.path,
+                    '--connect-timeout',
+                    '5',
```
Comment (theanalyst) [inline ceph_cfg/rados_client.py]: should this be the constant timeout which was introduced in the other pr?

Comment (oms4suse) [inline ceph_cfg/rados_client.py]: Absolutely, but cant do this until the other patch is in.

**Comment by oms4suse** [comment]
Ok fixed the constant timeout issue :)

**Comment by theanalyst** [comment]
ack

**Pull request merged**